### PR TITLE
Plane: move from FLAP_IN_CHANNEL to RCx_Option

### PR DESCRIPTION
**Pull request by IamPete1**
This moves from FLAP_IN_CHANNEL to an RC option.

I have done and tested the param convention, and it seems to work, but it could do with a check over by someone more experienced.

The point of this is so we can have flaps on boats too, https://github.com/ArduPilot/ardupilot/pull/11034

### CODE REVIEW

**Comment by IamPete1** [comment]
rebased

File: ArduPlane/Parameters.cpp [hunk 1]
```diff
@@ -1367,6 +1361,22 @@ void Plane::load_parameters(void)
...
+        Parameters::k_param_flapin_channel_old,
+        0,
+        AP_PARAM_INT8,
+        nullptr
+    };
+    if (AP_Param::find_old_parameter(&flap_info, &flap_output) && flap_output.get() != 0) {
+        RC_Channel *flapin = rc().channel(flap_output - 1);
+        if (!flapin->option.configured()) {
+            flapin->option.set_and_save((int16_t)RC_Channel::AUX_FUNC::FLAP); // save the new param
+            flap_output.set_and_save(0); // clear the old
```
Comment (tridge) [inline ArduPlane/Parameters.cpp]: don't do the set_and_save() here

File: ArduPlane/Parameters.cpp [hunk 2]
```diff
@@ -1367,6 +1361,22 @@ void Plane::load_parameters(void)
...
+    AP_Int8 flap_output;
+    AP_Param::ConversionInfo flap_info = {
+        Parameters::k_param_flapin_channel_old,
+        0,
+        AP_PARAM_INT8,
+        nullptr
+    };
+    if (AP_Param::find_old_parameter(&flap_info, &flap_output) && flap_output.get() != 0) {
+        RC_Channel *flapin = rc().channel(flap_output - 1);
+        if (!flapin->option.configured()) {
```
Comment (tridge) [inline ArduPlane/Parameters.cpp]: need to check for nullptr flapin

File: ArduPlane/Parameters.cpp (hunk 1 above)
Comment (IamPete1) [reply ArduPlane/Parameters.cpp]: fixed

File: ArduPlane/Parameters.cpp (hunk 2 above)
Comment (IamPete1) [reply ArduPlane/Parameters.cpp]: fixed

**Comment by IamPete1** [comment]
Rebased and fixed review issues but not yet re-tested

**Comment by IamPete1** [comment]
rebased

**Pull request merged**